### PR TITLE
FIX: prevents user to restore message deleted by staff

### PR DESCRIPTION
**Pull request by jjaffeux**
It could only occur on message created by the user itself and deleted while the user was looking at the channel.

It more generally fix the trash service which was not correctly setting the author of the delete.

### CODE REVIEW

File: plugins/chat/app/services/chat/trash_message.rb [hunk 1]
```diff
@@ -49,8 +49,8 @@ def invalid_access(guardian:, message:, **)
       guardian.can_delete_chat?(message, message.chat_channel.chatable)
     end
 
-    def trash_message(message:, **)
-      message.trash!
+    def trash_message(message:, guardian:, **)
+      message.trash!(guardian.user)
```
Comment (martin-brennan) [inline plugins/chat/app/services/chat/trash_message.rb:53]: Oh damn I do forget this a lot, thanks for fixing

File: plugins/chat/assets/javascripts/discourse/models/chat-message.js [hunk 2]
```diff
@@ -69,6 +70,7 @@ export default class ChatMessage {
     this.hidden = args.hidden || false;
     this.chatWebhookEvent = args.chatWebhookEvent || args.chat_webhook_event;
     this.createdAt = args.createdAt || args.created_at;
+    this.deletedById = args.deleted_by_id;
```
Comment (martin-brennan) [inline plugins/chat/assets/javascripts/discourse/models/chat-message.js:73]:
```suggestion
    this.deletedById = args.deletedById || args.deleted_by_id;
```

**Review by martin-brennan — APPROVED** [review]
Nice, LGTM

File: plugins/chat/assets/javascripts/discourse/models/chat-message.js (hunk 2 above)
Comment (jjaffeux) [reply plugins/chat/assets/javascripts/discourse/models/chat-message.js:73]: That actually shouldn't be needed (I know we have it just above), it shows something wrong if it is, but it doesn't hurt for now :+1:

**Pull request merged**